xcarchive/xcarchive.go

```diff
@@ -39,3 +42,38 @@ func IsMacOS(archPath string) (bool, error) {
 
 	return exist, nil
 }
+
+func unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, fileName string) (string, error) {
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("__xcarhive__")
+	if err != nil {
+		return "", err
+	}
+
+	if err := ziputil.UnZip(xcarchivePth, tmpDir); err != nil {
+		return "", err
+	}
+
+	applicationsPth := filepath.Join(tmpDir, filepath.Base(xcarchivePth), "Products", "Applications")
+	appName := strings.TrimSuffix(filepath.Base(xcarchivePth), filepath.Ext(xcarchivePth))
+
+	return utility.FindFileInAppDir(applicationsPth, appName, fileName)
+}
+
+// UnwrapEmbeddedMobileProvision ...
+func UnwrapEmbeddedMobileProvision(xcarchivePth string) (string, error) {
+	return unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, "embedded.mobileprovision")
+}
+
+// UnwrapEmbeddedInfoPlist ...
+func UnwrapEmbeddedInfoPlist(xcarchivePth string) (string, error) {
+	return unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, "Info.plist")
+}
+
+// CheckForXcarchive checks if the given zip is an xcarhive or not
+func CheckForXcarchive(pth string) bool {
+	filename := filepath.Base(pth)
```

Comment:
you no need to get the base of a file path, because the extension is going to be the suffix anyway, independently of the length of the path

Reply:
Removed.